core/src/main/java/com/linecorp/armeria/common/util/Sampler.java

```diff
@@ -80,7 +56,57 @@ public boolean isSampled() {
      *
      * @param rate minimum sample rate is 0.01, or 1% of traces
      */
-    public static Sampler create(float rate) {
+    static Sampler<Object> random(double rate) {
```

Comment:
I realized that this could be more useful to users if we did:

```java
static <T> Sampler<T> random(double rate) {
    @SuppressWarnings(...)
    final Sampler<T> cast = (Sampler<T>) CountingSampler.create(rate);
    return cast;
}
```